vm-presets.md

```diff
@@ -0,0 +1,185 @@
+Virtual Machine Presets
+=============================
+
+`VirtualMachinePresets` are an extension to general `VirtualMachine`
+Configuration based heavily on `PodPresets` from Kubernetes. When a
+`VirtualMachine` is created, any applicable `VirtualMachinePresets`
+will be applied to the existing spec for the `VirtualMachine`. This allows
+for re-use of common settings that should apply to multiple `VirtualMachines`.
+
+
+Implementation
+------------------------
+
+`VirtualMachinePresets` are implemented as a Kubernetes `Initializer`. This
+means the `VirtualMachine` resource is modified before it is visible to or
+processed by any other component of KubeVirt.
+
+One noteworthy difference between the behavior of a `VirtualMachinePreset` and
+a `PodPreset` is that if an error occurs during processing of a `PodPreset`,
+Kubernetes will continue to create the `Pod` without applying the `PodPreset`.
+KubeVirt will move the VirtualMachine to the `Failed` state instead. The
+rationale behind this difference is that settings defined in a
+`VirtualMachinePreset` are presumed to be critical to the function of the
+`VirtualMachine`.
+
+Application of `VirtualMachinePresets` is all-or-nothing. If an error occurs
+while a `VirtualMachinePreset` is being applied (for example, a conflict
+occurs), none of the `VirtualMachinePreset` will be applied (and the
+`VirtualMachine` will be put into a `Failed` state, as noted above).
+
+Once a `VirtualMachinePreset` is successfully applied to a `VirtualMachine`,
+the `VirtualMachine` will be marked with an annotation to indicate that is
+was applied.
+
+Usage
+------------------------
+
+Any domain structure can be listed in the spec of a `VirtualMachinePreset`.
```

Comment:
It might be worth adding that a preset can add elements into lists (lists inside devices and volumes for instance).

Reply:
Done!